packages/template-retail-react-app/app/components/store-locator-modal/store-locator-input.jsx

```diff
@@ -23,69 +23,114 @@ import {AlertIcon} from '@salesforce/retail-react-app/app/components/icons'
+                        }}
+                        render={({field}) => {
+                            return SUPPORTED_STORE_LOCATOR_COUNTRIES.length !== 0 ? (
+                                <FormControl isInvalid={form.formState.errors.countryCode}>
+                                    <Select
+                                        {...field}
+                                        marginBottom="10px"
+                                        placeholder={intl.formatMessage({
+                                            id: 'store_locator.action.select_a_country',
+                                            defaultMessage: 'Select a country'
+                                        })}
+                                        borderColor="gray.500"
+                                    >
+                                        {SUPPORTED_STORE_LOCATOR_COUNTRIES.map(
+                                            ({countryCode, countryName}) => {
+                                                return (
+                                                    <option value={countryCode} key={countryCode}>
+                                                        {intl.formatMessage(countryName)}
+                                                    </option>
+                                                )
+                                            }
+                                        )}
+                                    </Select>
+                                    {form.formState.errors.countryCode && (
+                                        <FormErrorMessage
+                                            sx={{marginBottom: '10px'}}
+                                            color="red.600"
+                                        >
+                                            <AlertIcon aria-hidden="true" mr={2} />
+                                            {form.formState.errors.countryCode.message}
+                                        </FormErrorMessage>
                                     )}
-                                </Select>
-                                {form.formState.errors.countryCode && (
-                                    <FormErrorMessage sx={{marginBottom: '10px'}} color="red.600">
-                                        <AlertIcon aria-hidden="true" mr={2} />
-                                        {form.formState.errors.countryCode.message}
-                                    </FormErrorMessage>
-                                )}
-                            </FormControl>
-                        ) : (
-                            <></>
-                        )
-                    }}
-                ></Controller>
+                                </FormControl>
+                            ) : (
+                                <></>
+                            )
+                        }}
+                    ></Controller>
+                ) : undefined}
```

Comment:
Minor: we can omit the `undefined` value here because React knows _not_ to render falsy values.

So it's ok to do something like this instead:
`{SUPPORTED_STORE_LOCATOR_COUNTRIES.length > 0 && <div>some content</div>`